lua/entities/starfall_prop/cl_init.lua

```diff
@@ -44,6 +44,17 @@ function ENT:GetRenderMesh()
 	end
 end
 
+function ENT:OnRemove()
+	-- This is required because snapshots can cause OnRemove to run even if it wasn't removed.
+	if self.rendermesh then
+		timer.Simple(0, function()
+			if not self:IsValid() then
+				self.rendermesh:Destroy()
```

Comment:
I just remembered this OnRemove might get called multiple times so we need `if self.rendermesh and not self:IsValid()` and also self.rendermesh = nil after the Destroy.